### PR TITLE
refactor(meta): some refactor on source/stream manager and fix recovery clean

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

In this PR:
1. add some cleaning for compaction group register and source manager, and pause source manager at the beginning of recovery.
2. removed background streaming job deleter, it's not necessary since we have already support batch drop streaming jobs, we can handle drop of streaming job right inside stream manager, ignore the failure and let recovery to take over the data cleaning.
3. some refactor in stream/source manager, including some rename, ie: rename `materialized view` with `streaming job`, because the stream and barrier manager will manage not only materialized views, but also sinks.
4. some refactor and fix on catalog manager, mainly about `drop_database`.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Review by shanicky — COMMENTED** [review]
Source/Scale part LGTM

**Comment by BugenZhao** [comment]
Will review it later. ❤️

**Comment by yezizp2012** [comment]
> Will review it later. ❤️

Just take your time, it's acceptable to review it next week. 😊

File: src/meta/src/manager/catalog/database.rs [hunk 1]
```diff
@@ -125,19 +126,21 @@ where
             in_progress_creation_tracker: HashSet::default(),
             in_progress_creation_streaming_job: HashSet::default(),
             in_progress_creating_tables: HashMap::default(),
+
+            _phantom: PhantomData::default(),
```
Comment (BugenZhao) [inline src/meta/src/manager/catalog/database.rs:130]:
```suggestion
            _phantom: PhantomData,
```

**Review by BugenZhao — APPROVED** [review]
LGTM

**Comment by BugenZhao** [comment]
This PR speeds up the e2e a lot on the CI. Interesting. 🤔 cc @xxhZs

**Pull request merged**